### ISSUE TITLE
rethinkDB java driver => ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by mrgren** [rethinkdb/rethinkdb#6058]
Hello
I am using : 
1 - jdk1.8.0_101
2 - glassfish 4.0
3- eclipse keepler with { 
--launcher.XXMaxPermSize 512M
--launcher.XXMaxPermSize 512m
-Xms40m
-Xmx1024m
}
4 -  	`<dependency` org="com.rethinkdb" name="rethinkdb-driver" `rev="2.3.1"/>`

on app boot i got i try to connect to My DB , but i got 

```
2016-08-16T16:14:22.940+0100|Severe: Exception while visiting com/rethinkdb/ErrorBuilder.class of size 5274  java.lang.ArrayIndexOutOfBoundsException: 3848

2016-08-16T16:14:22.991+0100|Severe: Exception while visiting com/rethinkdb/model/Arguments.class of size 3238
java.lang.ArrayIndexOutOfBoundsException: 8810


2016-08-16T16:14:22.993+0100|Severe: Exception while visiting com/rethinkdb/ast/ReqlAst.class of size 5974
java.lang.ArrayIndexOutOfBoundsException

2016-08-16T16:14:22.994+0100|Severe: Exception while visiting com/rethinkdb/ast/Query.class of size 4165
java.lang.ArrayIndexOutOfBoundsException: 52264

2016-08-16T16:14:22.995+0100|Severe: Exception while visiting com/rethinkdb/net/Connection.class of size 16143
java.lang.ArrayIndexOutOfBoundsException

2016-08-16T16:14:22.996+0100|Severe: Exception while visiting com/rethinkdb/net/Util.class of size 10480
java.lang.ArrayIndexOutOfBoundsException

2016-08-16T16:14:22.998+0100|Severe: Exception while visiting com/rethinkdb/net/SocketWrapper.class of size 6845
java.lang.ArrayIndexOutOfBoundsException

2016-08-16T16:14:22.999+0100|Severe: Exception while visiting com/rethinkdb/net/Cursor$DefaultCursor.class of size 3742
java.lang.ArrayIndexOutOfBoundsException: 52264


2016-08-16T16:14:23.000+0100|Severe: Exception while visiting com/rethinkdb/net/Cursor.class of size 7607
java.lang.ArrayIndexOutOfBoundsException

2016-08-16T16:14:23.001+0100|Severe: Exception while visiting com/rethinkdb/net/Response.class of size 8133
java.lang.ArrayIndexOutOfBoundsException

2016-08-16T16:14:23.001+0100|Severe: Exception while visiting com/rethinkdb/net/Converter.class of size 7326
java.lang.ArrayIndexOutOfBoundsException
```

any help plz?

**Comment by deontologician** [comment]
@mrgren What does your code look like that's throwing the error?

**Comment by mrgren** [comment]
sorry to be late 

this is my code
```
import com.rethinkdb.RethinkDB;
import com.rethinkdb.net.Connection;

public class ConnectionFactory {
	public static final RethinkDB r = RethinkDB.r;
	private static Connection conn = null;
	
	private ConnectionFactory(ServletContext context){
		String URI = MyURI;
		int PORT = MPort;
	     conn = r.connection().hostname(URI).port(PORT).connect();
	}
	
	public static Connection getInstance(ServletContext context){
		if(conn == null){
			new ConnectionFactory(context);
		}
		return conn;
	}
}

```

**Comment by bchavez** [comment]
I'm a bit out of my league here but what Java runtime does GlashFish officially support? From some glancing at the GlassFish docs looks like Java 7.

https://glassfish.java.net/getstarted.html

Is there a GlassFish version that supports Java 8?

Some **Google Fu** yields a similar exception trace:
http://www.adam-bien.com/roller/abien/entry/glassfish_v4_0_1_daily

AFAIK, you need at least Java 8 support for lambdas to use the Java RethinkDB driver.

---

:man: :checkered_flag: [***"Louie Louie, oh no... Me gotta go... Aye-yi-yi-yi, I said"***](https://www.youtube.com/watch?v=4V1p1dM3snQ)

**Comment by sdaguet** [comment]
From https://glassfish.java.net/download.html: "Java EE 7 requires JDK 7 or above, JDK 8 u60 or above is recommended for GlassFish 4.1.1"

From your link : "However, GlassFish v4.0.1 nightly builds builds starting with March, 20th 2014 supports Java 8 and lambdas. You can deploy Java EE 7 applications running on Java 8 with lambdas."

**Comment by mrgren** [comment]
Finally,  resolved  by using last glassFish Version 4.1.1
